File: src/main/java/epicsquid/superiorshields/enchantment/CapacityEnchantment.java
package epicsquid.superiorshields.enchantment;

import net.minecraft.world.item.enchantment.EnchantmentCategory;

public class CapacityEnchantment extends ShieldEnchantment {

	public CapacityEnchantment(Rarity rarityIn, EnchantmentCategory typeIn) {
		super(rarityIn, typeIn);
	}

	@Override
	public int getMaxLevel() {
		return 3;
	}

	@Override
	public int getMinCost(int pLevel) {
		return 5 + (pLevel - 1) * 8;
	}

	@Override
	public int getMaxCost(int pLevel) {
		return super.getMinCost(pLevel) + 50;
	}
}

File: src/main/kotlin/dev/epicsquid/superiorshields/enchantment/CapacityEnchantment.kt
package dev.epicsquid.superiorshields.enchantment

import dev.epicsquid.superiorshields.config.Config
import dev.epicsquid.superiorshields.effects.EffectHandler
import net.minecraft.world.item.enchantment.EnchantmentCategory

class CapacityEnchantment(
	rarity: Rarity,
	category: EnchantmentCategory,
	effectHandler: EffectHandler
) : SuperiorShieldEnchantment(rarity, category, effectHandler), AttributeProvider {
	override fun getMaxLevel(): Int = 3

	override fun getMinCost(pLevel: Int): Int = 5 + (pLevel - 1) * 8

	override fun getMaxCost(pLevel: Int): Int = super.getMinCost(pLevel) + 50

	override fun shieldAttributeModifiers(level: Int): ShieldAttributeModifiers =
		ShieldAttributeModifiers(
			capacity = level * Config.SHIELDS_CONFIG.shieldCapacityIncrease.get()
		)
}